pkg/scheduler/cache/node_info.go

```diff
@@ -380,6 +418,19 @@ func (n *NodeInfo) SetAllocatableResource(allocatableResource *Resource) {
 	n.generation = nextGeneration()
 }
 
+// GetGeneration returns the generation on this node.
+func (n *NodeInfo) GetGeneration() int64 {
+	if n == nil {
+		return 0
+	}
+	return n.generation
+}
+
+// SetGeneration sets the generation on this node.
```

Comment:
This needs a comment that it is only to be used for testing. I think we can get rid of this in a future cleanup.